File: Atcoder/keyence2020/D.cc
//同じものがあるとき間違える
#include<iostream>
#include<vector>
#include<algorithm>
#include<cmath>
#include<utility>
using namespace std;

signed main(){
  int n;cin >> n;
  vector<int> a(n);for(int i=0;i<n;i++) cin >> a[i];
  vector<int> b(n);for(int i=0;i<n;i++) cin >> b[i];
  int inf=10000000;
  int ans=inf;
  for(int i=0;i<(1<<n);i++){
    vector< vector<int> > x(n,vector<int>(3));
    for(int j=0;j<n;j++){
      if(i&(1<<j)){
        x[j][0]=a[j];x[j][1]=j;x[j][2]=0;
      }else{
        x[j][0]=b[j];x[j][1]=j;x[j][2]=1;
      }
    }
    vector< vector<int> > y(n,vector<int>(3));
    for(int j=0;j<n;j++){y[j]=x[j];}
    sort(x.begin(),x.end());
    bool f=true;
    //ありうる場合も除いてしまう
    for(int j=0;j<n;j++){
      if(abs(x[j][1]-j)%2!=x[j][2]){f=false;break;}
    }
    if(f){
      //ここ工夫する
      int c=0;
      for(int j=0;j<n-1;j++){
        for(int k=n-1;k>=j+1;k--){
          if(y[k]<y[k-1]){iter_swap(y.begin()+k,y.begin()+k-1);c+=1;}
          if(c==ans)break;
        }
        if(c==ans)break;
      }
      ans=c;
    }
  }
  if(ans==inf){
    cout << -1 << endl;
  }else{
    cout << ans << endl;
  }
}

File: Atcoder/keyence2020/D.py
n=int(input())
a=list(map(int,input().split()))
b=list(map(int,input().split()))
inf=100000000
ans=inf

for i in range(1<<n):#for i in range(2**n)と同じ
    x=[[a[j],j,0] if ((i >> j) & 1) == 1 else [b[j],j,1] for j in range(n)]
    y=[[x[j][0],x[j][1],x[j][2]] for j in range(n)]
    x.sort()
    for j in range(n):
        if abs(x[j][1]-j)%2!=x[j][2]:
            break
    else:
        cnt=0
        f=1 #隣接した要素が昇順になっていないものが存在する
        while f:
            f=0
            for k in range(n-1):
                if y[n-k-1]<y[n-k-2]:
                    y[n-k-1],y[n-k-2]=y[n-k-2],y[n-k-1]
                    f=1
                    cnt+=1
                if cnt>=ans:
                    break
            if cnt>=ans:
                    break
        else:
            ans=cnt
    #print(x)
    #print(ans)

if ans==inf:
    print(-1)
else:
    print(ans)
